dashboard/impl/src/java/org/sakaiproject/dash/logic/DashboardUserLogicImpl.java

```diff
@@ -108,19 +111,88 @@ public int countNewsLinksByGroupId(String sakaiUserId,
 							item.setTitle(typeObj.getGroupTitle(itemCount, item.getContext().getContextTitle(), item.getNewsTimeLabelKey()));
 						}
 					}
+				}else{
+					//When getItemCount() > 1 the infoLinkUrl is null and hence we don't need to run this call.
+					getInfoLinkUrlForNewsItem(item);
 				}
 			}
 		}
-		
+
 		return links;
 	}
 
+	private void getInfoLinkUrlForNewsItem(NewsItem item) {
```

Comment:
Changed to Setter and overloaded the method. Both methods has common logic but CalenderItem NewItem are two different object that don't share a common interface. hence Refactoring is not optimal.